interactive_markers/interactive_marker_server.py

```diff
@@ -138,9 +138,14 @@ def shutdown(self):
         can be destroyed.
         """
         self.clear()
-        self.applyChanges()
+        # It only makes sense to publish changes if the ROS context is still valid
+        if self.node.context.ok():
+            self.applyChanges()
+        self.node.destroy_service(self.get_interactive_markers_srv)
         self.get_interactive_markers_srv = None
```

Comment:
@jacobperron nit: it won't fail, but perhaps we shouldn't be passing `None` to `rclpy.Node.destroy_service()` if the server is shutdown twice (once explicitly, once before garbage collection):
```suggestion
        if self.get_interactive_markers_srv is not None:
            self.node.destroy_service(self.get_interactive_markers_srv)
            self.get_interactive_markers_srv = None
```
Same below.